### PR TITLE
SALTO-2819 - SalesForce: Create a filter that converts fields to lowercase

### PR DESCRIPTION
**Pull request by tomsellek**
Create a new filter that converts specific fields to lowercase

---

Some elements in SalesForce have names that are always provided as lower case, even if the name that's displayed in the SalesForce UI is mixed case. This causes problems when we want to create references to these elements, because the contents of the referring fields are mixed case. 
This PR creates a filter that changes said referring fields to lower case, so that we can easily change them into references.

---
_Release Notes_: 
SalesForce: The value of EntitlementTemplate.entitlementProcess field will now always appear as lower case, even if the EntitlementProcess it refers to has upper case letters in its name.

---
_User Notifications_: 
The value of EntitlementTemplate.entitlementProcess field will now always appear as lower case, even if the EntitlementProcess it refers to has upper case letters in its name.

### CODE REVIEW

**Comment by adamro** [comment]
I didn't think it through, but can't this cause issues with deploy? 
This is instance values, so I assume that when there's any change in the instance we deploy the whole instance.
Hence, when something else is changed in the instance, we will deploy with a lowercased value even tho the service value is not lowercased, causing a change the user did not do. No?

File: packages/salesforce-adapter/src/filters/convert_to_lowercase.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import { SALESFORCE } from '../constants'
+
+const { awu } = collections.asynciterable
+
+type FieldIdentifier = {
+  parentType: ElemID
+  fieldName: string
+}
+
+const fieldsToConvert: Array<FieldIdentifier> = [
```
Comment (Uri-Bar) [inline packages/salesforce-adapter/src/filters/convert_to_lowercase.ts:28]: I think you should take a look at how we did that in filters/convert_maps, beside the uniformity I think it is more readable. there is a lot of business logic that is very similar to what you did here

File: packages/salesforce-adapter/src/filters/convert_to_lowercase.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+const convertField = async (
+  instance: InstanceElement,
+  fieldIdentifierMap: Record<string, FieldIdentifier[]>
+): Promise<void> => {
+  const instanceElemId = (await instance.getType()).elemID.getFullName()
+  const fieldDescriptors = fieldIdentifierMap[instanceElemId]
+  if (!fieldDescriptors) {
+    return
+  }
+  fieldDescriptors.filter(fieldDesc => fieldDesc.fieldName in instance.value).forEach(fieldDesc => {
```
Comment (Uri-Bar) [inline packages/salesforce-adapter/src/filters/convert_to_lowercase.ts:44]: what will happened if the field you want to convert is not in the first layer of the values? in convert_maps we use more generic solution to my question and then more people will be able to use this filter

**Comment by tomsellek** [comment]
We'll do https://github.com/salto-io/salto/pull/3627 instead

**Pull request closed without merging**